src/plugins/workspace/public/components/workspace_picker_content/workspace_picker_content.tsx

```diff
@@ -88,24 +154,62 @@ export const WorkspacePickerContent = ({
       );
     });
     return (
-      <EuiListGroup showToolTips flush gutterSize="none" wrapText maxWidth={240}>
-        <EuiListGroupItem
-          label={
-            <EuiTitle size="xxs">
-              <h4>{itemType === 'all' ? allWorkspacesTitle : recentWorkspacesTitle}</h4>
-            </EuiTitle>
-          }
-        />
-        {listItems}
-      </EuiListGroup>
+      <>
+        <EuiTitle size="xxs">
+          <h4>{itemType === 'all' ? allWorkspacesTitle : recentWorkspacesTitle}</h4>
+        </EuiTitle>
+        <EuiSpacer size="s" />
+        <EuiListGroup showToolTips flush gutterSize="none" wrapText maxWidth={240}>
+          {listItems}
+        </EuiListGroup>
+        <EuiSpacer size="s" />
+      </>
     );
   };
 
   return (
     <>
-      {filteredRecentWorkspaces.length > 0 &&
-        getWorkspaceListGroup(filteredRecentWorkspaces, 'recent')}
-      {filteredWorkspaceList.length > 0 && getWorkspaceListGroup(filteredWorkspaceList, 'all')}
+      <EuiFlexGroup
+        direction="column"
+        justifyContent="center"
+        alignItems="center"
+        style={{ height: '100%' }}
+        gutterSize="none"
+      >
+        <EuiFlexItem grow={false} style={{ position: 'sticky', width: '100%' }}>
+          <EuiFieldSearch
+            compressed={true}
+            fullWidth={true}
+            value={search}
+            onChange={(e) => setSearch(e.target.value)}
+            placeholder={searchFieldPlaceholder}
+          />
+          <EuiSpacer />
+        </EuiFlexItem>
+
+        <EuiFlexItem grow={true} style={{ width: '100%', overflowY: 'auto' }}>
+          <EuiPanel paddingSize="none" color="transparent" hasBorder={false}>
+            {search ? (
+              <>
+                {queriedRecentWorkspace &&
+                  queriedRecentWorkspace.length > 0 &&
+                  getWorkspaceListGroup(queriedRecentWorkspace, 'recent')}
+                {queriedWorkspace &&
+                  queriedWorkspace.length > 0 &&
+                  getWorkspaceListGroup(queriedWorkspace, 'all')}
+                {(!queriedWorkspace || queriedWorkspace.length === 0) && getEmptyStatePrompt()}
+              </>
+            ) : (
+              <>
+                {filteredRecentWorkspaces.length > 0 &&
+                  getWorkspaceListGroup(filteredRecentWorkspaces, 'recent')}
+                {workspaceList.length > 0 && getWorkspaceListGroup(workspaceList, 'all')}
+                {workspaceList.length === 0 && getEmptyStatePrompt()}
+              </>
+            )}
```

Comment:
Can we remove the `search ? :`? From line 77, queriedRecentWorkspace = filteredRecentWorkspaces when there the search is empty. And I feel the variable name is a little bit confusing, do you think renaming filteredRecentWorkspaces -> recentWorkspaces makes more sense? Because from the first glance I thought the filteredRecentWorkspaces are the workspaces after searching.

Reply:
Hi, thanks for the useful comment, I have updated it.